components/Admin/Admin.js

```diff
@@ -22,35 +24,67 @@ class Admin extends React.Component {
 
   handleSearch = (event) => {
     const searchTerm = event.currentTarget.value
-    let { locations, filteredLocations } = this.state
+    let { locations, matchingSearchLocations } = this.state
 
     if (searchTerm) {
-      filteredLocations = locations.filter(location =>
+      matchingSearchLocations = locations.filter(location =>
```

Comment:
You're reassigning to a state field here. Not sure if this destructuring is creating a shallow copy or not. Can we instead do:

```js
const newSearchLocations = searchTerm ? locations.filter(...) : null
this.setState ({ matchingSearchLocations: newSearchLocations })
```